evals/compare_original_and_generated.py

```diff
@@ -0,0 +1,229 @@
+import os
+
+import torch
+import numpy as np
+import pretty_midi
+import pandas as pd
+import torch.nn as nn
+import fortepyan as ff
+from omegaconf import OmegaConf
+from datasets import load_dataset
+from torch.utils.data import DataLoader
+from fortepyan.audio import render as render_audio
+from huggingface_hub.file_download import hf_hub_download
+
+from sample import Generator
+from data.dataset import MidiDataset
+from data.quantizer import MidiQuantizer
+from models.reverse_diffusion import Unet
+from models.forward_diffusion import ForwardDiffusion
+from models.velocity_time_encoder import VelocityTimeEncoder
+
+
+def makedir_if_not_exists(dir: str):
+    if not os.path.exists(dir):
+        os.makedirs(dir)
+
+
+def preprocess_dataset(dataset_name: str, batch_size: int, num_workers: int):
+    ds = load_dataset(dataset_name, split="validation")
+    ds = MidiDataset(ds)
+    dataloader = DataLoader(ds, batch_size=batch_size, num_workers=num_workers, shuffle=True)
+
+    return dataloader
+
+
+def get_maestro_record(query: str | int):
+    ds = load_dataset("roszcz/maestro-v1-sustain", split="validation")
+    quantizer = MidiQuantizer(7, 7, 7)
+
+    if isinstance(query, int):
+        record = ds[query]
+    else:
+        idx_query = [i for i, record in enumerate(ds) if str.lower(query) in str.lower(f"{record['composer']} {record['title']}")]
+        # get first record
+        idx = idx_query[0]
+        record = ds[idx]
+
+    midi_filename = f"{record['composer']} {record['title']}"
+    piece = ff.MidiPiece.from_huggingface(record)
+
+    piece.df["next_start"] = piece.df.start.shift(-1)
+    piece.df["dstart"] = piece.df.next_start - piece.df.start
+
+    piece_quantized = quantizer.quantize_piece(piece)
+
+    sequence = {
+        "filename": midi_filename,
+        "pitch": torch.tensor(piece.df.pitch, dtype=torch.long)[:1024],
+        "dstart": torch.tensor(piece.df.dstart, dtype=torch.float)[:1024],
+        "duration": torch.tensor(piece.df.duration, dtype=torch.float)[:1024],
+        "velocity": (torch.tensor(piece.df.velocity, dtype=torch.float)[None, None, :1024] / 64) - 1,
+        "dstart_bin": torch.tensor(piece_quantized.df.dstart_bin, dtype=torch.long)[None, :1024],
+        "duration_bin": torch.tensor(piece_quantized.df.duration_bin, dtype=torch.long)[None, :1024],
+        "velocity_bin": torch.tensor(piece_quantized.df.velocity_bin, dtype=torch.long)[None, :1024],
+    }
+
+    return sequence
+
+
+def denormalize_velocity(velocity: np.ndarray):
+    return (velocity + 1) * 64
```

Comment:
When running this eval I get an error:

```
Traceback (most recent call last):
...
  File "/home/hagrid/workson/midi-diffusion/evals/compare_original_and_generated.py", line 150, in compare_original_and_generated
    render_midi_to_mp3(model_midi, f"tmp/mp3/{filename}-model.mp3")
  File "/home/hagrid/workson/midi-diffusion/evals/compare_original_and_generated.py", line 160, in render_midi_to_mp3
    render_audio.midi_to_mp3(midi_file, filepath)
...
    raise ValueError('data byte must be in range 0..127')
ValueError: data byte must be in range 0..127
```

I'm not sure if we have to clip this, or is there some deeper issue here?